src/wp-includes/abilities-api/class-wp-abilities-category-registry.php

```diff
@@ -0,0 +1,258 @@
+<?php
+/**
+ * Abilities API
+ *
+ * Defines WP_Abilities_Category_Registry class.
+ *
+ * @package WordPress
+ * @subpackage Abilities API
+ * @since 6.9.0
+ */
+
+declare( strict_types = 1 );
+
+/**
+ * Manages the registration and lookup of ability categories.
+ *
+ * @since 6.9.0
+ * @access private
+ */
+final class WP_Abilities_Category_Registry {
+	/**
+	 * The singleton instance of the registry.
+	 *
+	 * @since 6.9.0
+	 * @var self|null
+	 */
+	private static $instance = null;
+
+	/**
+	 * Holds the registered categories.
+	 *
+	 * @since 6.9.0
+	 * @var WP_Ability_Category[]
+	 */
+	private $registered_categories = array();
+
+	/**
+	 * Registers a new category.
+	 *
+	 * Do not use this method directly. Instead, use the `wp_register_ability_category()` function.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @see wp_register_ability_category()
+	 *
+	 * @param string              $slug The unique slug for the category. Must contain only lowercase
+	 *                                  alphanumeric characters and dashes.
+	 * @param array<string,mixed> $args {
```

Comment:
Nit: in all instances of `array<string,mixed>` or `array<string,string>` (and so on) a space should be added after the comma.